opm/material/fluidmatrixinteractions/EclMaterialLawManager.hpp

```diff
@@ -868,6 +947,51 @@ class EclMaterialLawManager
         }
     }
 
+    template <class Container>
+    void readGasWaterEffectiveParameters_(Container& dest,
+                                        const Opm::EclipseState& eclState,
+                                        unsigned satRegionIdx)
+    {
+        if (!hasGas || !hasWater)
+            // we don't read anything if either the gas or the water phase is not active
+            return;
+
+        dest[satRegionIdx] = std::make_shared<GasWaterEffectiveTwoPhaseParams>();
+
+        auto& effParams = *dest[satRegionIdx];
+
+        const auto& tableManager = eclState.getTableManager();
+
+        switch (eclState.runspec().saturationFunctionControls().family()) {
+        case SatFuncControls::KeywordFamily::Family_I:
+        {
+            // No Family I applicable in this case
+            break;
+        }
+
+        case SatFuncControls::KeywordFamily::Family_II:
+        {
+            //Todo: allow also for Sgwfn table input as alternative to Sgfn and Swfn table input
+            const SgfnTable& sgfnTable = tableManager.getSgfnTables().getTable<SgfnTable>( satRegionIdx );
+            const SwfnTable& swfnTable = tableManager.getSwfnTables().getTable<SwfnTable>( satRegionIdx );
+
+            std::vector<double> SwColumn = swfnTable.getColumn("SW").vectorCopy();
+            
+            effParams.setKrwSamples(SwColumn, swfnTable.getColumn("KRW").vectorCopy());
+            effParams.setKrnSamples(SwColumn, sgfnTable.getColumn("KRG").vectorCopy());
```

Comment:
Please use [`normalizeKrValues()`](https://github.com/OPM/opm-material/blob/ae3791b2b013e124e122d4c6d8274b895828970d/opm/material/fluidmatrixinteractions/EclMaterialLawManager.hpp#L993-L1004) here to account for `TOLCRIT`.  See PR #439 (commit bc38dcb189158e6575e2cb694093d32c5f74f772) for additional details.

Reply:
> It looks like you're missing the end-point scaling feature for gas-water. Do you plan to add that in a future PR?

That I will do in a future PR.